shared/modules/persistence_module.go

```diff
@@ -55,15 +55,12 @@ type PersistenceWriteContext interface {
 	// Block Operations
 
 	// Indexer Operations
-	// TODO(#315): Change `txResult TxResult` to `txBytes []byte`
-	StoreTransaction(txResult TxResult) error
+	StoreTransactions() error
 
 	// Block Operations
-	// TEMPORARY: Including two functions for the SQL and KV Store as an interim solution
-	//                 until we include the schema as part of the SQL Store because persistence
-	//                 currently has no access to the protobuf schema which is the source of truth.
-	StoreBlock(blockProtoBytes []byte) error                                              // Store the block in the KV Store
-	InsertBlock(height uint64, hash string, proposerAddr []byte, quorumCert []byte) error // Writes the block in the SQL database
+	SetLatestTxResults(txResults []TxResult)
```

Comment:
I'm beginning to lose context of what `last` and `latest` mean throughout the codebase, and understand I also added to this confusion in different parts of the code.

Do you think it makes sense to use `SetProposalTxResults` here instead so it's clear it is the current proposal and not yet committed? Ditto for the primitives backing this.

Reply:
> I'm beginning to lose context of what last and latest mean throughout the codebase, and understand I also added to this confusion in different parts of the code.

Understood - will take your suggestion below

> Do you think it makes sense to use SetProposalTxResults here instead so it's clear it is the current proposal and not yet committed? Ditto for the primitives backing this.

See comment above about `ApplyBlock` being used during state sync